### ISSUE TITLE
Implementing the CLI

### ISSUE THREAD

**Issue by lafemmephile** [lafemmephile/mosaic#17]
Something like this:
```python
import os
import click


def block_range_check(ctx, param, value):
    # check range format: min max
    if value[0] > value[1]:
        ctx.fail(
            f"Invalid value for '{param.opts[0]}' / '{param.opts[1]}':" +
            " " +
            f"range format {value} is not (small_block, large_block)."
        )

    # otherwise
    return value


@click.group(invoke_without_command=True, no_args_is_help=True)
@click.version_option()
@click.option('-i', '--interactive', is_flag=True, help='Drop into interactive prompt mode.')
def main(interactive: bool) -> None:
    """Generate an image/vidoe mosaic based on gradient and bvh subdivision."""
    if interactive:
        welcome_text = "Welcome to use this library. This library aims to produce an image mosaic that preserves detail and tries to use all the images provided. Good luck have fun!"
        print(welcome_text)
        MainMenu()


@main.command('prepare', no_args_is_help=True, short_help='Prepare an image/video folder for mosaic generation.')
@click.option('-c', '--color-file-name', 'cfname', type=click.Path(), default='color_info.json', show_default=True, help='JSON file containing color mappings for use in mosaic generation.')
@click.option('-f', '--frame-skip', 'skip', type=int, default=30, show_default=True, help='# of video frames to skip between samples.')
@click.option('-o', '--frame-save-dir', type=click.Path(), default=os.getcwd(), show_default='current directory', help='Path to write video frame samples to.')
@click.option('-p', '--tile-path', type=click.Path(exists=True, dir_okay=True), default=os.getcwd(), show_default='current directory', help='Path of images/videos directory for mosaic generation.')
@click.argument('tile_type', type=click.Choice(['image', 'video'], case_sensitive=False))
def prepare(cfname: str, skip: int, frame_save_dir: str, tile_path: str, tile_type: str) -> None:
    """Prepare an image/video folder for generation, all images/videos in subfolders will be included."""
    pass


@main.command('generate', no_args_is_help=True, short_help='Generate a mosaic.')
@click.option('-z', '--resize', type=click.IntRange(1), default=1, show_default=True, help='Used to enlarge the target image.')
@click.option('-r', '--block-range', type=click.Tuple([click.IntRange(1), click.IntRange(1)]), callback=block_range_check, default=(100, 200), show_default=True, help='Block range used in mosaic: (min, max).')
@click.option('-a', '--alpha', type=click.FloatRange(0, 1, min_open=True, max_open=True), default=0.5, show_default=True, help='Used with gradient value (from image) to calculate subdivision frequency.')
@click.argument('target', type=click.Path(exists=True, file_okay=True))
@click.argument('color_file', type=click.Path(exists=True, file_okay=True))
def generate(resize: int, block_range: tuple[int, int], alpha: float, target: str, color_file: str) -> None:
    """Generate a mosaic from a given target image file and a prepared JSON color file."""
    pass
```

References:
+ https://click.palletsprojects.com/en/8.1.x/options/
+ https://click.palletsprojects.com/en/8.1.x/commands/
+ https://click.palletsprojects.com/en/8.1.x/prompts/

**Comment by lafemmephile** [comment]
Additional References:
+ https://click.palletsprojects.com/en/8.1.x/api/#parameters

**Comment by lafemmephile** [comment]
Need to implement:
+ https://github.com/lafemmephile/mosaic/blob/d52a048b550f79e20bac31f471eb0ed312cf5c72/src/mosaic/__main__.py#L309-L323
+ https://github.com/lafemmephile/mosaic/blob/d52a048b550f79e20bac31f471eb0ed312cf5c72/src/mosaic/__main__.py#L370-L384

These `sub commands` are simply the *non-interactive* version of their *interactive* counterparts (i.e. options [1-2], and 3 from the interactive main menu).

**Comment by lafemmephile** [comment]
Implemented with: e0e501469b34fe4ec16547f7144988262c964bf9
